File: apollo/src/main/java/io/muun/apollo/domain/errors/InvalidPaymentRequestError.java
package io.muun.apollo.domain.errors;


import io.muun.apollo.external.UserFacingErrorMessages;
import io.muun.common.exception.PotentialBug;

public class InvalidPaymentRequestError extends UserFacingError implements PotentialBug {

    public InvalidPaymentRequestError(String innerMessage) {
        this(innerMessage, null);
    }

    /**
     * Strange constructor with innerMessage instead of just message.
     */
    public InvalidPaymentRequestError(String innerMessage, Throwable cause) {
        super(
                UserFacingErrorMessages.INSTANCE.invalidPaymentRequest(),
                new RuntimeException(innerMessage, cause)
        );
    }
}

File: android/apollo/src/main/java/io/muun/apollo/domain/errors/InvalidPaymentRequestError.kt
package io.muun.apollo.domain.errors


import io.muun.apollo.data.external.UserFacingErrorMessages
import io.muun.common.exception.PotentialBug

class InvalidPaymentRequestError
    @JvmOverloads constructor(innerMessage: String, cause: Throwable? = null): UserFacingError(
        UserFacingErrorMessages.INSTANCE.invalidPaymentRequest(),
        RuntimeException(innerMessage, cause)
    ), PotentialBug
